projects/plugins/jetpack/class-jetpack-stats-dashboard-widget.php

```diff
@@ -0,0 +1,161 @@
+<?php
+/**
+ * Adds the Jetpack stats widget to the WordPress admin dashboard.
+ *
+ * @package jetpack
+ */
+
+use Automattic\Jetpack\Assets\Logo as Jetpack_Logo;
+use Automattic\Jetpack\Status;
+
+/**
+ * Class that adds the Jetpack stats widget to the WordPress admin dashboard.
+ */
+class Jetpack_Stats_Dashboard_Widget {
+
+	/**
+	 * Indicates whether the class initialized or not.
+	 *
+	 * @var bool
+	 */
+	private static $initialized = false;
+
+	/**
+	 * Initialize the class by calling the setup static function.
+	 *
+	 * @return void
+	 */
+	public static function init() {
+		if ( ! self::$initialized ) {
+			self::$initialized = true;
+			self::wp_dashboard_setup();
+		}
+	}
+
+	/**
+	 * Sets up the Jetpack Stats widget in the WordPress admin dashboard.
+	 */
+	public static function wp_dashboard_setup() {
+		if ( Jetpack::is_connection_ready() ) {
+			add_action( 'jetpack_dashboard_widget', array( __CLASS__, 'dashboard_widget_footer' ), 999 );
+		}
+
+		if ( has_action( 'jetpack_dashboard_widget' ) ) {
+			$jetpack_logo = new Jetpack_Logo();
+			$widget_title = sprintf(
+				// translators: Placeholder is a Jetpack logo.
+				__( 'Stats by %s', 'jetpack' ),
+				$jetpack_logo->get_jp_emblem( true )
+			);
+
+			// Wrap title in span so Logo can be properly styled.
+			$widget_title = sprintf(
+				'<span>%s</span>',
+				$widget_title
+			);
+
+			wp_add_dashboard_widget(
+				'jetpack_summary_widget',
+				$widget_title,
+				array( __CLASS__, 'dashboard_widget' )
+			);
+			wp_enqueue_style( 'jetpack-dashboard-widget', plugins_url( 'css/dashboard-widget.css', JETPACK__PLUGIN_FILE ), array(), JETPACK__VERSION );
+			wp_style_add_data( 'jetpack-dashboard-widget', 'rtl', 'replace' );
+
+			// If we're inactive and not in offline mode, sort our box to the top.
```

Comment:
It's not clear to me why we'd bother reordering the widget? Am I missing something, or can we just drop this entire block.

If users have reordered their widgets it sounds like it won't even do anything anyways:

> ...this only works for people who have never re-ordered their widgets. Once a user has done so their existing preferences will override this and they will have to move your widget to the top for it to stay there.
via https://developer.wordpress.org/apis/handbook/dashboard-widgets/#forcing-your-widget-to-the-top

Reply:
Yeah I mentioned that [in this comment as well](https://github.com/Automattic/jetpack/pull/20676#discussion_r708630721). I can't find a scenario where the code is run anyway. I think we can nix it.

Reply:
@sdixon194 yeah, go ahead and drop that block and I can give it a final look.

Reply:
removed in 7a45952